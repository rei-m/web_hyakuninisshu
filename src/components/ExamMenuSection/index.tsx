import * as React from 'react';
import styled from 'styled-components';
import { withAppTheme } from '@src/styles';
import PageTitle from '@src/components/PageTitle';

export interface ExamMenuSectionProps {
  readonly onSubmit: () => void;
}

const RootSection = withAppTheme(styled.section)`
  max-width: 380px;
  margin: auto;
  box-sizing: border-box;
  padding: ${({ theme }) => theme.spacing2x};
`;

const Explain = withAppTheme(styled.div)`
  margin: ${({ theme }) => `${theme.spacing4x} ${theme.spacing2x}`};
`;

const StartButton = withAppTheme(styled.button)`
  margin-top: ${({ theme }) => theme.spacing4x};
  background-color: ${({ theme }) => theme.colorAccent} !important;
  &:active {
    background-color: ${({ theme }) => theme.colorAccentActive} !important;
  }
  &:hover {
    background-color: ${({ theme }) => theme.colorAccentHover} !important;
  }
`;

const ExamMenuSection: React.SFC<ExamMenuSectionProps> = ({ onSubmit }) => (
  <RootSection>
    <PageTitle title="腕試し" />
    <div>
      <Explain>
        全百首からランダムに出題されます。
        <br />
        練習の成果を確認しましょう。
      </Explain>
      <StartButton
        type="submit"
        className="pt-button pt-intent-primary pt-large pt-icon-edit"
        onClick={onSubmit}
        data-test="start-exam-button"
      >
        腕試しをはじめる
      </StartButton>
    </div>
  </RootSection>
);

export default ExamMenuSection;
