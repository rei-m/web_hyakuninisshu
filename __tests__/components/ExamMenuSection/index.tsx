import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { shallow } from 'enzyme';
import ExamMenuSection from '../../../src/components/ExamMenuSection';
import { sel } from '../../helpers';

describe('<ExamMenuSection />', () => {
  it('should render component', () => {
    const renderer = ReactTestRenderer.create(
      <ExamMenuSection onSubmit={jest.fn()} />
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should fire onSubmit when start button clicked', () => {
    const mockHandler = jest.fn();
    const wrapper = shallow(<ExamMenuSection onSubmit={mockHandler} />);
    wrapper.find(sel('start-exam-button')).simulate('click');
    expect(mockHandler).toHaveBeenCalled();
  });
});
