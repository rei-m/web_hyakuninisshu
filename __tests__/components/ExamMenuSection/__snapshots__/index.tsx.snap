// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExamMenuSection /> should render component 1`] = `
<section
  className="sc-bwzfXH gRkoHi"
>
  <h1
    className="sc-bdVaJa fXsTJn"
  >
    腕試し
  </h1>
  <div>
    <div
      className="sc-htpNat kJuGSE"
    >
      全百首からランダムに出題されます。
      <br />
      練習の成果を確認しましょう。
    </div>
    <button
      className="pt-button pt-intent-primary pt-large pt-icon-edit sc-bxivhb cJoDeg"
      data-test="start-exam-button"
      onClick={[MockFunction]}
      type="submit"
    >
      腕試しをはじめる
    </button>
  </div>
</section>
`;
